### ISSUE TITLE
Add a "maintainer" tag in  the meta.yaml

### ISSUE THREAD

**Issue by ChrisBarker-NOAA** [conda-forge/conda-smithy#12]
I think it would be a good idea to have a tag in  each recipe's meta.yaml file that indicates where the package is maintained. the recipe travels with the built package, so this would be a way for anyone that got the package from who knows where to know where to go to report bugs, issues etc.

I understand conda now supports a "extras" section. So we could do something like:

```
extras:
   - recipe_maintainer : https://github.com/conda-forge/udunits-feedstock
```

**Comment by pelson** [comment]
Perhaps we should actually associate github handles...

```
extras:
   recipe_maintainers:
       - pelson
       - ChrisBarker-NOAA
```

Ideally, any major change to a recipe (compiler flags etc.) should be merged by one of the people mentioned in the maintainers list, though we also have gh teams to do this kind of thing more rigidly.

**Comment by mwcraig** [comment]
Could we populate that list from the people with administrative rights on the feedstock repo? That would be one less moving part to keep current...

**Comment by ChrisBarker-NOAA** [comment]
from the perspective of the third-party user -- I think simply the gitHub repo is all they need to know.

On the other hand, if there are multiple people with write privileges to the repo, then we may want to sort out who's responsible for what -- but I'm not sure that needs to be in meta.yaml

**Comment by jakirkham** [comment]
+1

**Comment by pelson** [comment]
I've started the ball rolling by putting 
```
extra:
  recipe-maintainers:
    - pelson
```
in https://github.com/conda-forge/cycler-feedstock

My intention is for conda-forge teams to be managed based on this metadata. i.e. there will be a feedstock crawler which manages the github teams with merge permissions on the appropriate recipes.

**Comment by ChrisBarker-NOAA** [comment]
My idea, at least, was that we wantd a way for fols that found the package in teh wild -- or even on anaconda-org, to know where it came from, and therefor where they could ask questions / report bugs, so maybe:
```
extra:
  recipe-maintainers:
    - https://github.com/conda-forge/cycler-feedstock : pelson
```

In fact, I don't think you need the name -- the repo is really the info needed. If someone goes to that repo, they will see who the maintainer(s) are, and be able to post issues.

Do we want folks hunting us down and contacting us outside that repo anyway?

I suppose we could call it "recipe-repo" instead, though what I was hoping was that this could become a convention used outside of conda-forge. -- Ideally every recipe (and thus every package) would have this tag.

-CHB

**Comment by jakirkham** [comment]
I kinda like @pelson's idea. Looks a bit cleaner. Though it would be nice to have the URL somewhere. Maybe as separate dictionary in `extra`.

    extra:
      recipe-url: https://github.com/conda-forge/cycler-feedstock
      recipe-maintainers:
        - pelson

**Comment by ChrisBarker-NOAA** [comment]
well, just the name is no enough -- that only works if you already know this package came from the gitHub conda-forge organization.

but sure, recipe-url is a good idea.

-CHB

**Comment by pelson** [comment]
The conda-smithy linter now checks for a recipe-maintainers section. (#42)

**Comment by pelson** [comment]
I'm tempted by the idea of a ```feedstock-url``` section:

```
extra:
  feedstock-url: https://github.com/conda-forge/cycler-feedstock
  recipe-maintainers:
    - pelson
```

The main motivation would be for forked recipes to be tracked back to their original home. That said, if this is about tracking back from anaconda.org, we have the ability to control the URL/description/summary on those pages, and I absolutely agree that they should be trackable back to the recipe itself.

**Comment by JanSchulz** [comment]
But then it should be "original-conda-recipe" or so... Feedstock is a bit odd as forking the feedstock means changing the URL. That would only makes sense if this bit of information comes up in the final package ("Oh, this package is from there...").

**Comment by ocefpaf** [comment]
:+1:  to keep maintainer as is.  (People can choose to put their e-mail or github username.)
:+1:  to recipe URL.  That will increase recipe noise, but I believe it is worth it.

> Feedstock is a bit odd as forking the feedstock means changing the URL.

IMO it should.  People should update the URL to reflect the fork.

> That would only makes sense if this bit of information comes up in the final package .

In a way it will be there.  Conda does save the recipe in the package.

**Comment by jakirkham** [comment]
@pelson @JanSchulz why not just `recipe-url` for the name? This makes it more straightforward and `conda`-like.

**Comment by jakirkham** [comment]
Another thought just given that `extra` may attract lots of other stuff.

    extra:
      recipe:
          url: https://github.com/conda-forge/cycler-feedstock
          maintainers:
            - pelson

**Comment by JanSchulz** [comment]
Re URL: IMO there should be a way to bake the URL into the final package by getting it from git or whatever (it should be available on the package builder, as they have to checkout the package+branch?). Having it in the source recipe (which I know where it comes from) is IMO going to ask for bitrot. Duplicated data is never good :-( 

Idea: monkey patch in a jinja function, which gets this information during the build process? Like getting it from the environment variables in travis/circleci/appveyor and if these are not set, use git to ask for the current remote?

```
$ git config branch.`git name-rev --name-only HEAD`.remote
origin
$ git remote get-url origin
https://github.com/bliker/cmder.git
```

Re maintainer: I've similar feelings about this. If it is a single recipe feedstock (like conda smithy is implemented right now), then having this information in the recipe is not giving the user much. At least not until there is an easy way to see this information (which I currently know no way: `conda list` only shows channels). If it is a multiple recipe repo (like staged-recipe), then this makes sense...

re github handle vs email: seems that github won the "social network for code" war, so thumbs up :-) Maybe adding it as `github:@handle` or so?

**Comment by pelson** [comment]
I think we've talked this one through enough. The recipe maintainers section is working really nicely now.

The one area that this issue touches on which hasn't yet been refined is the ability to track back to the feedstock repo based on the distribution installed. I'm happy to address that as a separate issue, if there is a compelling usecase.

Thanks all! :+1:

**Comment by jakirkham** [comment]
The badges (added to the Readmes) will help with this by letting the user link back to the source. Though I know this isn't the same as having it in the metadata.

It is worth keeping in mind that all conda-forge stuff is in one channel. So, maybe what we need is to have a summary line that says from conda-forge as opposed to a link per se. Just a thought. ;)
